discovery-provider/src/tasks/entity_manager/delegate.py

```diff
@@ -0,0 +1,127 @@
+import json
+import logging
+from typing import Optional, TypedDict, Union, cast
+
+from src.models.delegates.delegate import Delegate
+from src.tasks.entity_manager.utils import Action, EntityType, ManageEntityParameters
+from src.utils.indexing_errors import EntityMissingRequiredFieldError
+from src.utils.model_nullable_validator import all_required_fields_present
+
+logger = logging.getLogger(__name__)
+
+
+class CreateDelegateMetadata(TypedDict):
+    address: Union[str, None]
+    name: Union[str, None]
+    is_personal_access: Union[bool, None]
+
+
+def get_create_delegate_metadata_from_raw(
+    raw_metadata: Optional[str],
+) -> Optional[CreateDelegateMetadata]:
+    metadata: CreateDelegateMetadata = {
+        "address": None,
+        "name": None,
+        "is_personal_access": None,
+    }
+    if raw_metadata:
+        try:
+            json_metadata = json.loads(raw_metadata)
+            raw_address = json_metadata.get("address", None)
+            if raw_address:
+                metadata["address"] = raw_address.lower()
+            else:
+                metadata["address"] = None
+            metadata["name"] = json_metadata.get("name", None)
+            metadata["is_personal_access"] = json_metadata.get(
+                "is_personal_access", None
+            )
+            return metadata
+        except Exception as e:
+            logger.error(
```

Comment:
Could we use the new logging context stuff @isaacsolo added?

Reply:
yea feel free to modify the logger to use `StructuredLogger` like this https://github.com/AudiusProject/audius-protocol/blob/main/discovery-provider/src/tasks/index_nethermind.py#L86